File: routes/equity.js
import { Router } from 'express'
import { func, utils, schema, get, add } from 'ynp'
import { db } from '../db.js'

const router = Router()

router.get('/balance', async (req, res) => {
	const parsed = schema.Balance.safeParse(req.body)
	if (!parsed.success) {
		return res.status(400).json({ result: parsed.error.format() })
	}

	try {
		const { user_id } = parsed.data
		const auuid = await get.user(BigInt(user_id), db)
		const result = await get.balance(auuid, db)
		res.status(200).json({ result })
	} catch (err) {
		res.status(404).json({ result: err.message })
	}
})

router.post('/pay', async (req, res) => {
	const parsed = schema.TransferCoin.safeParse(req.body)
	if (!parsed.success) {
		return res.status(400).json({ result: parsed.error.format() })
	}

	try {
		const { sender_id, receiver_id, amount } = parsed.data
		const result = await func.transferCoin( sender_id, receiver_id, amount, db )
		res.status(200).json({ result })
	} catch (err) {
		res.status(err.status || 404).json({ result: err.message })
	}
})

router.post('/eval', async (req, res) => {
	const parsed = schema.Eval.safeParse(req.body)
	if (!parsed.success) {
		return res.status(400).json({ result: parsed.error.format() })
	}

	try {
		const { user_id, message_id, message_length, timestamp } = parsed.data
		const result = await func.evalDiscord( user_id, message_id, message_length, timestamp, db )
		res.status(200).json({ result })
	} catch (err) {
		res.status(404).json({ result: err.message })
	}
})

router.delete('/del', async (req, res) => {
	const parsed = schema.Del.safeParse(req.body)
	if (!parsed.success) {
		return res.status(400).json({ result: parsed.error.format() })
	}

	try {
		const { message_id } = parsed.data
		const result = await func.delDiscord( message_id, db )
		res.status(200).json({ result })
	} catch (err) {
		res.status(404).json({ result: err.message })
	}
})

export default router

File: routes/equity.py
from fastapi import APIRouter

from pkg import DiscordUsers
from pkg import database
from pkg import schemas
from pkg import fun

router = APIRouter()
DB = database.DB

@router.get('/')
async def equity():
  return { 'msg': 'equity' }

# discord
@router.post('/discord/eval', response_model=schemas.Response)
async def discord_eval(req: schemas.Eval, db: DB) -> schemas.Response:
  user = fun.User(DiscordUsers, req.platform_id, db)
  return await user.evalMessage(req.message_length)

@router.get('/discord/balance', response_model=schemas.Response)
async def balance(req: schemas.Balance, db: DB) -> schemas.Response:
  user = fun.User(DiscordUsers, req.platform_id, db)
  return await user.balance()

@router.post('/discord/pay', response_model=schemas.Response)
async def pay(req: schemas.Pay, db: DB) -> schemas.Response:
  user = fun.User(DiscordUsers, req.sender_platform_id, db)
  return await user.pay(req)

# minecraft
# @router.post('/minecraft/eval', response_model=schemas.Response)
# async def minecraft_eval(req: schemas.Eval, db: DB) -> schemas.Response:
# 	return await fun.eval(req, MinecraftUsers, db)

# @router.get('/minecraft/balance', response_model=schemas.Response)
# async def balance(req: schemas.Balance, db: DB) -> schemas.Response:
# 	return await fun.balance(req, db)

# @router.post('/minecraft/pay', response_model=schemas.Response)
# async def pay(req: schemas.Pay, db: DB) -> schemas.Response:
# 	return await fun.pay(req, db)
